default/python/universe_generation/galaxy.py

```diff
@@ -205,16 +217,50 @@ def __init__(self, positions, adjacency_grid):
         self.dsets = DisjointSets()
 
         for pos in positions:
-            self.add(pos)
+            self._add(pos)
 
-    def add(self, pos):
+        self.clusters = set()
+        for cluster in self.dsets.complete_sets():
+            self.clusters.add((len(cluster), frozenset(cluster)))
+
+    def __len__(self):
+        return len(self.clusters)
+
+    def _add(self, pos):
+        """Adds pos."""
         for neighbor in self.adjacency_grid.neighbors(pos):
             # print "Clusterer adding ", pos,  " and ", neighbor
             self.dsets.link(pos, neighbor)
 
-    def clusters(self):
-        """Returns a list of lists of the clusters of positions"""
-        return self.dsets.complete_sets()
+    def smallest_isolated_cluster(self):
+        """
+        Return None if all of the positions are in one cluster
+        or the smallest cluster
+        """
+        return None if len(self.clusters) < 2 else min(self.clusters)[1]
+
+    def stitch_clusters(self, p1, p2, stitches):
```

Comment:
I will change the comment to use the phrase "stitch together" because it evokes the visual image of two clusters stitched with a path of systems, which is exactly how it looks with clusters more than one star apart.